app/views/CreditAndRefunds.scala.html

```diff
@@ -82,162 +64,199 @@
     }
 }
 
+@creditAndRefundRefundProgressHtml(amount: BigDecimal) = {
+    @if(messages.lang.language.toLowerCase().equals("cy")) {
+        <li>
+            <p class="govuk-heading-s">
+                <span class="govuk-body">@messages("credit-and-refund.refundProgress-prt-1")</span>
+                @amount.toCurrencyString
+                <span class="govuk-body">@messages("credit-and-refund.refundProgress-prt-2")</span>
+            </p>
+        </li>
+    } else {
+        <li>
+            <p class="govuk-heading-s">@amount.toCurrencyString
+                <span class="govuk-body">@messages("credit-and-refund.refundProgress")</span>
+            </p>
+        </li>
+    }
+}
+
+@maybeCreditChargesHead = @{
+    creditCharges.headOption.getOrElse(throw MissingFieldException("Credit Charges"))
+}
+
+@maybeBalancesDetails = @{
+    balance.getOrElse(throw MissingFieldException("Balance"))
+}
+
+@maybeAvailableCredit = @{
+    maybeBalancesDetails.availableCredit.getOrElse(throw MissingFieldException("Available Credit"))
+}
+
```

Comment:
this maybe worth to move all these conditions/checks on the controller level?
I think adding some sort of mode variable of custom type/enum [None / UnallocatedCreditFromOnePayment/ isUnallocatedCreditFromSingleCreditItem] would work better;

Reply:
Good idea thank you. I've just pushed the refactored code